src/AstroImages.jl

```diff
@@ -97,13 +109,22 @@ end
 Construct an `AstroImage` object of `data`, using `color` as color map, `Gray` by default.
 """
 AstroImage(color::Type{<:Color}, data::Matrix{T}, wcs::WCSTransform) where {T<:Real} =
-    AstroImage{T,color, 1}((data,), (wcs,))
-AstroImage(color::Type{<:Color}, data::NTuple{N, Matrix{T}}, wcs::NTuple{N, WCSTransform}) where {T<:Real, N} =
-    AstroImage{T,color, N}(data, wcs)
-AstroImage(data::Matrix{T}) where {T<:Real} = AstroImage{T,Gray,1}((data,), (WCSTransform(2),))
-AstroImage(data::NTuple{N, Matrix{T}}) where {T<:Real, N} = AstroImage{T,Gray,N}(data, ntuple(i-> WCSTransform(2), N))
-AstroImage(data::Matrix{T}, wcs::WCSTransform) where {T<:Real} = AstroImage{T,Gray,1}((data,), (wcs,))
-AstroImage(data::NTuple{N, Matrix{T}}, wcs::NTuple{N, WCSTransform}) where {T<:Real, N} = AstroImage{T,Gray,N}(data, wcs)
+    AstroImage{T,color, 1, AbstractFloat}((data,), (wcs,), Properties{AbstractFloat}())
+function AstroImage(color::Type{<:Color}, data::NTuple{N, Matrix{T}}, wcs::NTuple{N, WCSTransform}) where {T<:Real, N}
+    if N == 3 && color == RGB && T <: Union{AbstractFloat, FixedPoint}
+        img = ccd2rgb((data[1], wcs[1]),(data[2], wcs[2]),(data[3], wcs[3]))
+        return AstroImage{T,color,N, widen(T)}(data, wcs, Properties{widen(T)}(rgb_image = img))
+    elseif N == 3 && color == RGB
+        img = ccd2rgb((data[1], wcs[1]),(data[2], wcs[2]),(data[3], wcs[3]))
+        return AstroImage{T,color,N, AbstractFloat}(data, wcs, Properties{AbstractFloat}(rgb_image = img))
```

Comment:
Why the parameter is abstract here?

Reply:
@giordano The parameter is Abstract here for the following reasons: 
- When we fixed the `MappedArrays` type for `rgb_image`, the RGB subtype in it can only have `P` as an `AbstractFloat` or `FixedPoint` numbers.
- Also, note that we are not filling `rgb_image` in all cases. So, when I simply passed `T` as a parameter (as done in the previous commit), it generated errors when `T` was not `AbstractFloat` or `FixedPoint`.
- A solution for this was to pass a valid parameter just for the sake of passing it (when `rgb_image` is not getting filled). That's what is done except in one case where `rgb_image` is filled and the type isn't a valid one. The type needs to be in the above-mentioned domain as per `Images.jl`.